runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -249,7 +249,7 @@ public void onPlayerDeath(PlayerDeath playerDeath)
 		{
 			takeScreenshot("Death", "Deaths");
 		}
-		else if ((player.isClanMember() || player.isFriend()) && config.screenshotFriendDeath() && player.getCanvasTilePoly() != null)
+		else if ((player != client.getLocalPlayer() || player.isClanMember() || player.isFriend()) && config.screenshotFriendDeath() && player.getCanvasTilePoly() != null)
```

Comment:
This logic is incorrect; the reason the bug happens is due to `isFriend()` returning `true` for the local player. This is a behavior of the underlying Jagex client (I think so that you can see your own chat with chat set to friends?). It may be worth adding a test for the issue described into `ScreenshotPluginTest`.

Reply:
Would comparing names work?

Reply:
You don't need to compare names. Right now, the issue with your code is that `player != client.getLocalPlayer() || ... || player.isFriend()` is always true.